src/cli.rs

```diff
@@ -126,10 +127,19 @@ impl Cli {
                 force_handle.stop()?;
                 Ok(())
             }
+            // Handle the metrics server stopping
+            result = metrics_handle => {
+                if let Err(e) = result {
+                    error!("Metrics server crashed: {}", e);
+                    Err(eyre::eyre!("Metrics server stopped unexpectedly, shutting down..."))
+                } else {
+                    Ok(())
+                }
```

Comment:
Since `init_metrics_server` invokes an infinite loop, is there any time that we will exit early and return an error here?